userspace/libsinsp/cri.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright (C) 2013-2019 Draios Inc dba Sysdig.
+
+This file is part of sysdig.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "cri.h"
+
+#include "sinsp.h"
+#include "sinsp_int.h"
+
+namespace libsinsp {
+namespace cri {
+std::string s_cri_unix_socket_path = "/run/containerd/containerd.sock";
+std::unique_ptr <runtime::v1alpha2::RuntimeService::Stub> s_cri = nullptr;
+int64_t s_cri_timeout = 1000;
+sinsp_container_type s_cri_runtime_type = CT_CRI;
+
+sinsp_container_type get_cri_runtime_type(const std::string &runtime_name)
+{
+	if(runtime_name == "containerd")
+	{
+		return CT_CONTAINERD;
+	} else
+	{
+		return CT_CRI;
+	}
+}
+
+bool parse_cri_image(const runtime::v1alpha2::ContainerStatus &status, sinsp_container_info *container)
+{
+	// image_ref may be one of two forms:
+	// host/image@sha256:digest
+	// sha256:digest
+	const auto &image_ref = status.image_ref();
+	auto digest_start = image_ref.find("sha256:");
```

Comment:
I know it's a corner case, but wouldn't this also match images like `myregistry/image_name_that_ends_with_sha256:mytag`?
Or since tags never end up in `image_ref()` you're sure that there can only 1 occurence of `:`?